### PR TITLE
add support for buf

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: buf.yaml
Comment (johanbrandhorst) [inline buf.yaml:2]: This won't work until this org and repository is in place, do you want to create it and take ownership of it?

File: .github/workflows/buf.yaml
Comment (johanbrandhorst) [inline .github/workflows/buf.yaml:16]: Who ever owns the org has to configure this secret on this repository for this push to work

File: buf.yaml
Comment (remyleone) [reply buf.yaml:2]: I've asked @roidelapluie to create directly as he is a prometheus maintainer.

Comment (johanbrandhorst) [reply buf.yaml:2]: FYI, I've reserved this under out bot user for now, please reply here or reach out on our Slack if you want to take over ownership of this organization. Also CC @brancz, @bwplotka, they might be interested in this effort too.

**Comment by beorn7** [comment]
This looks very neat, but I'm missing a bit of context. Could you provide a bit of explanation? Or perhaps you already talked to  @roidelapluie , @brancz , or @bwplotka about this and they have an idea about this from the perspective of Prometheus?

**Comment by johanbrandhorst** [comment]
FYI, to support https://buf.build/envoyproxy/envoy, Buf has made https://buf.build/prometheus/client-model a managed third party library on the BSR, so there's no need for this unless you're interested in adopting the maintenance of the library. I think this PR can be closed.

**Comment by roidelapluie** [comment]
Ideally we (Prometheus-team) would own the Prometheus organization on buf.build and publush using this pull request. Same as the remote write Pr's. How can we transfer ownership?

**Comment by johanbrandhorst** [comment]
> Ideally we (Prometheus-team) would own the Prometheus organization on buf.build and publush using this pull request. Same as the remote write Pr's. How can we transfer ownership?

Wonderful! Is there somewhere I can reach one of you for more direct discussion? You can find me on the Gophers slack or the official Buf slack: https://buf.build/links/slack.

**Comment by beorn7** [comment]
Thank you very much @roidelapluie for being on top of things. I leave handling of this PR to you then. :-)

File: buf.work.yaml
Comment (johanbrandhorst) [inline buf.work.yaml:3]: I think we can remove this file. The existing managed repository on buf.build uses the root of the repository as the protobuf root: https://buf.build/prometheus/client-model/tree.

File: .github/workflows/buf.yaml
Comment (johanbrandhorst) [reply .github/workflows/buf.yaml:16]: @roidelapluie this would be you :).

File: buf.yaml
Comment (johanbrandhorst) [reply buf.yaml:2]: We'll want to update this to point to the managed repo for a smooth transition: `buf.build/prometheus/client-model`. The BSR does not allow `_` in repository names.

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
I'm in the process of handing over the `prometheus` org on https://buf.build to @roidelapluie, and we'll need to make the suggested changes before we can merge this.

**Comment by johanbrandhorst** [comment]
Oh and you may want to add a `buf.md` to override the one in the current repository which says that it is managed by buf.  For more information see our docs: https://docs.buf.build/bsr/documentation#module-documentation.

**Comment by remyleone** [comment]
@johanbrandhorst done I've added the content of the readme.

File: .github/workflows/buf.yaml
Comment (johanbrandhorst) [inline .github/workflows/buf.yaml:4]: I think we want to run the lint and breaking change detector on all PRs, and just limit `push` to the `master` branch.

**Review by johanbrandhorst — COMMENTED** [review]
Thanks for the update, this looks almost done to me!

**Comment by bboreham** [comment]
This looks to have stalled, was there a change of plan?

**Comment by brancz** [comment]
Honestly I’m not sure we still have to do this (although I continue to be a big fan of the buf tooling).

FWIW we can already and I already have used remote plugin execution to consume promtheus’ protos. Eg.: https://github.com/polarsignals/profile-exporter/blob/67df95d8b88629894c0c5b7ebdb4bf77ec5dc763/cmd/profile-exporter/remote_write.go#L25

**Comment by beorn7** [comment]
I have no strong preference. In the past, @roidelapluie seems to have been driving this. Not sure what's the current state.

Obviously, there is the know messiness that we have a proto3 version of the same exposition format over in prometheus/prometheus, see https://github.com/prometheus/prometheus/blob/main/prompb/io/prometheus/client/metrics.proto , and maybe a published version of this should use the more modern proto3.

And maybe maybe it's anyway better to wait for OM protobuf to become a real thing and advertise that one instead of this classic Prometheus protobuf format.

**Comment by johanbrandhorst** [comment]
Buf maintains a copy of this on the BSR, I don't think we need to add anything here unless we want to modernize the generation strategy.

**Comment by beorn7** [comment]
@roidelapluie could you make a call what to do here?

**Comment by roidelapluie** [comment]
I am going to close this until there is evidence that someone needs us to pick this.

**Pull request closed without merging**
